### PR TITLE
Data jpa

### PR DESCRIPTION
**Pull request by YevheniiVlasenko**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -2,5 +2,7 @@ spring.datasource.url=jdbc:mysql://localhost/test?serverTimezone=UTC
 spring.datasource.username=root
 spring.datasource.password=715914qQ()
```
Comment (AgroFix) [inline src/main/resources/application.properties:3]: I hope this is not your real password😏

File: src/main/java/project/bookstore/service/BookService.java [hunk 2]
```diff
@@ -3,11 +3,16 @@
...
     BookDto save(CreateBookRequestDto requestDto);
 
     List<BookDto> findAll();
 
-    public BookDto getBookById(Long id);
+    BookDto getBookById(Long id);
+
+    void safeDelete(Long id);
+
+    Book update(Long id, CreateBookRequestDto createBookRequestDto);
```
Comment (AgroFix) [inline src/main/java/project/bookstore/service/BookService.java]: why Book, but not BookDto

**Review by AgroFix — APPROVED** [review]

File: src/main/java/project/bookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -35,4 +35,22 @@ public BookDto getBookById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException(GET_BY_ID_MESSAGE)));
     }
+
+    @Override
+    public void safeDelete(Long id) {
```
Comment (kshuryhin) [inline src/main/java/project/bookstore/service/BookServiceImpl.java:40]: I guess more general method name, like `delete` will be preferable here. In my opinion, if in the future you would like to change safe delete on real one, those implementation details will not affect more abstract service method name. But that is minor point

File: src/main/java/project/bookstore/service/BookService.java (hunk 2 above)
Comment (kshuryhin) [inline src/main/java/project/bookstore/service/BookService.java]: It looks like we should operate with DTOs on the service layer

File: src/main/java/project/bookstore/model/Book.java [hunk 4]
```diff
@@ -5,11 +5,17 @@
...
+import jakarta.persistence.Table;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Data
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id =?")
+@Where(clause = "is_deleted=false")
```
Comment (kshuryhin) [inline src/main/java/project/bookstore/model/Book.java]: 'org.hibernate.annotations.Where' is deprecated

**Review by kshuryhin — CHANGES_REQUESTED** [review]
Good job, but there are some points to consider

File: src/main/resources/application.properties (hunk 1 above)
Comment (YevheniiVlasenko) [reply src/main/resources/application.properties:3]: just another mock data)

File: src/main/java/project/bookstore/model/Book.java (hunk 4 above)
Comment (YevheniiVlasenko) [reply src/main/java/project/bookstore/model/Book.java]: It is, but it's used in the lesson video

**Review by vovasalyha — APPROVED** [review]
LGTM

**Pull request merged**